src/libclipper/src/rpc_service.cpp

```diff
@@ -141,16 +145,34 @@ void RPCService::manage_service(const string address) {
       // TODO: Balance message sending and receiving fairly
       // Note: We only receive one message per event loop iteration
       log_info(LOGGING_TAG_RPC, "Found message to receive");
-
       receive_message(socket, connections, connections_containers_map,
                       zmq_connection_id, redis_connection);
+      auto current_time = std::chrono::system_clock::now();
+      std::chrono::duration<double> time_pass_since_last_check = current_time - last_check_time_;
+      if(time_pass_since_last_check.count() > CONTAINER_EXISTENCE_CHECK_FREQUENCY){
+         check_containers_existence(receiving_history_);
+         last_check_time_ = current_time;
+      }
     }
     // Note: We send all queued messages per event loop iteration
     send_messages(socket, connections);
   }
   shutdown_service(socket);
 }
 
+void RPCService::check_containers_existence(std::map<const vector<uint8_t>, std::chrono::system_clock::time_point> receiving_history_){
```

Comment:
This method doesn't need to take the receiving history as a parameter. `receiving_history_` is a member variable of the `RPCService` class and can be referenced directly in the method body.